src/components/TransactionHistoryRows/utils.ts

```diff
@@ -12,14 +12,16 @@ export const getTxMetadataWithAssetId = (txMetadata?: TxMetadata) => {
   if (txMetadata && 'assetId' in txMetadata) return txMetadata
 }
 
-export const getDisplayTransfers = (transfers: Transfer[], types: TransferType[]): Transfer[] => {
-  return types.reduce<Transfer[]>((prev, type) => {
-    const transfer = transfers.find(t => t.type === type)
-    if (!transfer) return prev
-    prev.push(transfer)
+export const getTransfersByType = (
+  transfers: Transfer[],
+  types: TransferType[],
+): Record<TransferType, Transfer[]> =>
+  types.reduce((prev, type) => {
+    const transfersOfType = transfers.filter(t => t.type === type)
+    if (!transfersOfType.length) return prev
+    prev[type] = [...(prev[type] ?? []), ...transfersOfType]
     return prev
```

Comment:
we shouldn't need the spreading since we are doing a full filter on type now
```suggestion
    const transferByType = transfers.filter(t => t.type === type)
    if (!transferByType.length) return prev
    prev[type] = transferByType
    return prev
```